data/src/emulator.js

```diff
@@ -7174,5 +7174,51 @@ class EmulatorJS {
 
         return recorder;
     }
+
+    enableSaveUpdateEvent() {
+        if (!window.crypto.subtle) {
+            console.error("Attempt to compare save files without subtle crypto (maybe you are running in an insecure context?");
+            return;
+        }
+
+        // https://stackoverflow.com/questions/9267899/
+        function transformDigestToBase64(buffer) {
+            var binary = '';
+            var bytes = new Uint8Array( buffer );
+            var len = bytes.byteLength;
+            for (var i = 0; i < len; i++) {
+                binary += String.fromCharCode( bytes[ i ] );
+            }
+            return window.btoa( binary );
+        }
+
+        function withGameSaveHash(saveFile, callback) {
+            if (saveFile) {
+                window.crypto.subtle.digest("SHA-256", saveFile).then(digest => callback(transformDigestToBase64(digest), saveFile));
```

Comment:
I don't think we need SHA-256 for seeing if a file changed. sha1 should be fine - or if you want higher, maybe this should be configurable